src/parsers/withdrawals.ts

```diff
@@ -91,11 +90,19 @@ function parseStakerWithdrawalEvent<T>(ev: T): DepoWithdrawEvent {
     return event;
 }
 
-function get3CRVUSDAmount(coinAmount: BigInt, tx:string): BigInt{
-    const price = get3CrvVirtualPrice(tx);
-    const result = coinAmount.times(price);
-    return result.div(BigInt.fromI32(10) // remove price's decimals
-                       .pow(18));
+function getUSDAmountOfShare(tokenIndex: number, coinAmount:BigDecimal): BigInt {
+    let usdAmount = BigDecimal.zero();
+    if(tokenIndex == 0){
+        usdAmount = amountToUsd("dai", coinAmount);
+    } else if(tokenIndex == 1){
+        usdAmount = amountToUsd("usdc", coinAmount);
+    } else if(tokenIndex == 2) {
+        usdAmount = amountToUsd("usdt", coinAmount);
+    } else {
```

Comment:
A withdrawal without index is always 3crv? could there be another case? In general, I would use the last else to set the value to `0` if no tokenIndex was found.

Reply:
Well, normally things can break when the theory/assumptions clash with an edge case 😌